feat: add app setting page

windforce852 opened the pull request:
Main Feature:
- implemented 'language' drop down, connected with a sample data array.
- implemented 'theme' drop down, with two staic options 'light', 'dark'
- drop down were implemented with react-hook-form
- added AppSetting page into routeConfig
- added entry from side nav bar
- added unit tests that passed after migrate to next
- changed theme hook to fit next js

<img width="1051" alt="Screenshot 2024-06-12 at 6 33 04 PM" src="https://github.com/BeeQuantAI/platform_app/assets/84086904/73ad8415-db67-48d2-8f98-477c7e063d46">

<img width="1045" alt="Screenshot 2024-06-12 at 6 32 57 PM" src="https://github.com/BeeQuantAI/platform_app/assets/84086904/278dbc26-346c-4d36-bbce-039ac6317e94">

<img width="1048" alt="Screenshot 2024-06-12 at 6 33 13 PM" src="https://github.com/BeeQuantAI/platform_app/assets/84086904/2aea155e-2ccb-4287-ac83-5c1f7c0f8902">

Chris-Wang reviewed (changes requested):
Please try to solve the ui mismatch issue in safari:
<img width="1700" alt="image" src="https://github.com/BeeQuantAI/platform_app/assets/1610136/e69d345a-19b9-4372-808b-a821194115f8">

Chris-Wang commented on src/app/(protected)/appsetting/page.tsx:
```diff
@@ -0,0 +1,119 @@
+                    />
+                  </FormGroupField>
+                </FormGroup>
+
```
No need to leave between lines in tsx return part

windforce852 replied:
do you mean `/>` in line 76 should merge to the tail of 75? done

Chris-Wang commented on src/app/(protected)/appsetting/page.tsx:
```diff
@@ -0,0 +1,119 @@
+  FormGroupLabel,
+} from '@/shared/components/form/FormElements';
+import { Controller, useForm } from 'react-hook-form';
+
```
need to remove empty lines here

windforce852 commented:
ready to rebase after most updated dev issue solved

Commit:
feat: add app setting page

Added AppSetting page:
- implemented 'language' drop down, with Eng and Chn language options
- implemented 'theme' drop down, with two staic options 'light', 'dark'
- drop down were wrote by custom ul li, implemented within react-hook-form
- added AppSetting page into routeConfig
- added entry from side nav bar
- added unit tests that passed after migrate to next
- changed theme hook to fit next js

Resolve CP-36

windforce852 commented:
Pushed a new verion that included the custom dropdown that using 'ul' 'li' rather than 'select' 'option' ; and added related test
